pkg/controllerhelpers/wait_test.go

```diff
@@ -0,0 +1,75 @@
+package controllerhelpers
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func hasKeyFunc(name string) func(*corev1.ConfigMap) (bool, error) {
+	return func(cm *corev1.ConfigMap) (bool, error) {
+		_, found := cm.Data[name]
+		return found, nil
+	}
+}
+
+func TestAggregatedConditions_Condition(t *testing.T) {
+	t.Parallel()
+	obj := &corev1.ConfigMap{
+		Data: map[string]string{
+			"alpha": "foo",
+			"beta":  "bar",
+		},
+	}
+
+	tt := []struct {
+		name           string
+		ac             *AggregatedConditions[*corev1.ConfigMap]
+		expectedDone   bool
+		expectedString string
+		expectedErr    error
+	}{
+		{
+			name:           "single done condition is done",
+			ac:             NewAggregatedConditions[*corev1.ConfigMap](hasKeyFunc("alpha")),
+			expectedDone:   true,
+			expectedString: "[true]",
+			expectedErr:    nil,
+		},
+		{
+			name:           "done+undone condition is undone",
+			ac:             NewAggregatedConditions[*corev1.ConfigMap](hasKeyFunc("alpha"), hasKeyFunc("doesnotexist")),
+			expectedDone:   false,
+			expectedString: "[true,false]",
+			expectedErr:    nil,
+		},
+		{
+			name:           "done+undone+done condition is undone",
+			ac:             NewAggregatedConditions[*corev1.ConfigMap](hasKeyFunc("alpha"), hasKeyFunc("doesnotexist"), hasKeyFunc("beta")),
+			expectedDone:   false,
+			expectedString: "[true,false,true]",
+			expectedErr:    nil,
+		},
```

Comment:
I'd also add a test for missing state (`<nil>`)

Reply:
added